src/lib.rs

```diff
@@ -87,11 +100,35 @@ impl<T: BitAlloc> BitAlloc for BitAllocCascade16<T> {
             None
         }
     }
-    fn dealloc(&mut self, key: usize) {
+
+    fn alloc_contiguous_at(
+        &mut self,
+        base: usize,
+        size: usize,
+        align_log2: usize,
+    ) -> Option<usize> {
+        if check_contiguous(self, base, Self::CAP, size, align_log2) {
+            self.remove(base..base + size);
+            Some(base)
+        } else {
+            None
+        }
+    }
+
+    fn dealloc(&mut self, key: usize) -> bool {
         let i = key / T::CAP;
-        self.sub[i].dealloc(key % T::CAP);
         self.bitset.set_bit(i, true);
+        self.sub[i].dealloc(key % T::CAP)
+    }
+
+    fn dealloc_contiguous(&mut self, base: usize, size: usize) -> bool {
+        let mut success = true;
+        for key in base..base + size {
```

Comment:
yep, the final operation can be done through `self.insert(range)`

however, I think we need to traverse this range bit by bit to check if this range is allocated contiguously, 

why not just use the `dealloc` which is also completed bit-by-bit

Reply:
Because I think use `dealloc` bit-by-bit needs to iterate from the upper layer to the lower layer for each bit, and this iterative process feels like a repetitive operation that will bring performance overhead